File: DependencyInjection/DependencyInjection/_1_ControlFreak/Services/MsSqlBasedRepository.cs
using DependencyInjectionBefore._1_ControlFreak.Interfaces;

namespace DependencyInjectionBefore._1_ControlFreak.Services
{
  class MsSqlBasedRepository
  {
    private readonly DataDestination _sqlDataDestination = new SqlDataDestination();

    public void Save(AcmeMessage message)
    {
      message.WriteTo(_sqlDataDestination);
    }
  }
}

File: Java/dependency-injection/control-freak/src/main/java/com/github/grzesiek_galezowski/DependencyInjection/_1_ControlFreak/Services/MsSqlBasedRepository.java
package com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Services;

import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Interfaces.AcmeMessage;
import com.github.grzesiek_galezowski.DependencyInjection._1_ControlFreak.Interfaces.DataDestination;

public class MsSqlBasedRepository {
  private final DataDestination _sqlDataDestination = new SqlDataDestination();

  public void save(AcmeMessage message) {
    message.writeTo(_sqlDataDestination);
  }
}
